MobileWeather/ImageProcessor.m

```diff
@@ -32,8 +32,8 @@ - (instancetype)init {
 }
 
 - (UIImage *)imageFromConditionImage:(NSString *)conditionImage {
```

Comment:
To fix the blurriness, you should add a parameter here for an enum of "image size", large (primary graphic) == 256 (very crisp on TDK), small (for soft buttons / menu / choice set) == 64? Maybe 128? Try it out and see what works. You can also pass a weight (smaller items should be weighted thicker?)

Below comment describes what to do with these parameters.